projects/solar/test/test_sense_mcp3427.c

```diff
@@ -176,7 +176,7 @@ void test_sense_mcp3427_normal_cycle_one_mcp3427(void) {
   TEST_ASSERT_EQUAL(false, is_set);  // not setting data yet
 
   // call the MCP3427 callback, make sure nothing was set yet (we only set data on the sense cycle)
-  s_mcp3427_callbacks[0](TEST_SENSED_CH1_VALUE, TEST_SENSED_CH2_VALUE,
+  s_mcp3427_callbacks[0](TEST_SENSED_CH1_VALUE, (int16_t)TEST_SENSED_CH2_VALUE,
```

Comment:
I'm not really sure why you're making unrelated changes in power_selection and solar? Especially since these seem to be redundant. In general unrelated changes should be submitted as a separate PR - e.g. if we have to revert a commit it's better that it reverts one thing and not a bunch of unrelated changes that we have to put back.